doc/plugins.dox

```diff
@@ -358,78 +457,70 @@ Once this is done, your first plugin is ready. Congratulations!
 @section listing Complete listing (without comments)
 
 @code
-#include <geanyplugin.h>
-
-GeanyPlugin		*geany_plugin;
-GeanyData		*geany_data;
 
-PLUGIN_VERSION_CHECK(211)
+#include <geanyplugin.h>
 
-PLUGIN_SET_INFO("HelloWorld", "Just another tool to say hello world",
-				"1.0", "John Doe <john.doe@example.org>");
+static gboolean hello_init(GeanyPlugin *plugin, gpointer pdata)
+{
+	GtkWidget *main_menu_item = gtk_menu_item_new_with_mnemonic("Hello World");
+	gtk_widget_show(main_menu_item);
 
+	geany_plugin_set_data(plugin, main_menu_item, NULL);
+}
 
-static GtkWidget *main_menu_item = NULL;
 
-static void item_activate_cb(GtkMenuItem *menuitem, gpointer gdata)
+static void hello_cleanup(GeanyPlugin *plugin, gpointer pdata)
 {
-	dialogs_show_msgbox(GTK_MESSAGE_INFO, "Hello World");
-}
+	GtkWidget *main_menu_item = (GtkWidget *) pdata;
 
-void plugin_init(GeanyData *data)
-{
-	main_menu_item = gtk_menu_item_new_with_mnemonic("Hello World");
-	gtk_widget_show(main_menu_item);
-	gtk_container_add(GTK_CONTAINER(geany->main_widgets->tools_menu),
-		main_menu_item);
-	g_signal_connect(main_menu_item, "activate",
-		G_CALLBACK(item_activate_cb), NULL);
+	gtk_widget_destroy(main_menu_item);
 }
 
-void plugin_cleanup(void)
+
+G_MODULE_EXPORT
+void geany_load_module(GeanyPlugin *plugin, GModule *module, gint geany_api_version)
 {
-	gtk_widget_destroy(main_menu_item);
+	plugin->info->name = "HelloWorld";
+	plugin->info->description = "Just another tool to say hello world";
+	plugin->info->version = "1.0";
+	plugin->info->author = "John Doe <john.doe@example.org>";
+
+	plugin->funcs->init = hello_init;
+	plugin->funcs->cleanup = hello_cleanup;
+
+	GEANY_PLUGIN_REGISTER(plugin, 225);
 }
- @endcode
+@endcode
 
 
 Now you might like to look at Geany's source code for core plugins such as
 @a plugins/demoplugin.c.
 
-@section furtherimprovements Furter Improvements and next steps
+@section furtherimprovements Further Improvements and next steps
 @subsection translatable_plugin_information Translatable plugin information
 
 After having written our first plugin, there is still room for improvement.
 
-By default, PLUGIN_SET_INFO() does not allow translation of the basic plugin
+By default, geany_load_module() does does not allow translation of the basic plugin
```

Comment:
"allow" is kind of wrong now.  We don't allow it or not, we just don't setup a custom catalog for the user.  With `PLUGIN_SET_INFO()` it was different as it expanded to a whole function so couldn't easily be expanded.

"As is, none of the strings are translated.  To enable translations, you need to first setup Gettext with main_locale_init(), before any usual calls to gettext's \c _()."

And if we really want to speak about core plugins, add "Calling main_locale_init() is not required from plugins shipped with Geany as they share the translation catalog which is already setup by Geany itself".